errbot/backends/irc.py

```diff
@@ -146,10 +148,14 @@ def __repr__(self):
 
 
 class IRCRoom(Room):
+
     def __init__(self, room, bot):
         self._bot = bot
         self.room = room
         self.connection = self._bot.conn.connection
+        self._topic = None
+        # TODO: Check if this leaks a room on reconnect
+        self._bot._rooms[self.room] = self
```

Comment:
You have a race condition here. `self._bot._rooms` may be modified concurrently by different threads.